CharacterBuilder/src/app/api.service.ts

```diff
@@ -0,0 +1,20 @@
+import {firstValueFrom, map} from 'rxjs';
+import {ImageData} from './build/build.component';
+import {HttpClient} from '@angular/common/http';
+import {inject, Injectable} from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ApiService {
+  private readonly httpClient = inject(HttpClient);
+
+  async createImage(imageData: ImageData) {
+    return await firstValueFrom(this.httpClient.post('http://localhost:5110/build-image-url', imageData));
```

Comment:
Try to avoid duplicating constants like the server name of the API. Put the base URL `http://localhost:5110` in a const and use it multiple times.

Even better would be defining an injection token in _app.config_ (see e.g. https://github.com/rstropek/htl-2024-25-4AHIF/blob/main/homeworks/CharacterBuilder/ui/src/app/app.config.ts). We did not discuss that during lectures, but maybe you want to take a look at it.